### PR TITLE
filter out referenced toc from output

### PR DESCRIPTION
**Pull request by OsmondJiang**
1. filter out referenced toc from output
2. optimize test case

### CODE REVIEW

File: src/docfx/build/Build.cs [hunk 1]
```diff
@@ -96,6 +96,12 @@ private static bool ShouldBuildFile(Document itemToBuild, ConcurrentDictionary<D
                 return false;
             }
 
+            if (itemToBuild.ContentType == ContentType.TableOfContents &&
```
Comment (yufeih) [inline src/docfx/build/Build.cs]: Can this be done in BuildToc instead of here?

Comment (OsmondJiang) [reply src/docfx/build/Build.cs]: I think we should control the build item in one place

---
In reply to: [186925294](https://github.com/dotnet/docfx/pull/2745#discussion_r186925294) [](ancestors = 186925294)

File: src/docfx/build/toc/TableOfContentsMap.cs [hunk 2]
```diff
@@ -11,16 +11,23 @@ namespace Microsoft.Docs.Build
...
     {
-        private readonly List<Document> _tocs;
+        private readonly HashSet<Document> _tocs;
 
         private readonly Dictionary<Document, HashSet<Document>> _documentToTocs;
 
         public TableOfContentsMap(List<Document> tocs, Dictionary<Document, HashSet<Document>> documentToTocs)
         {
-            _tocs = tocs ?? throw new ArgumentNullException(nameof(tocs));
+            _tocs = new HashSet<Document>(tocs ?? throw new ArgumentNullException(nameof(tocs)));
```
Comment (vwxyzh) [inline src/docfx/build/toc/TableOfContentsMap.cs]: debug.assert instead of throw?

Comment (vwxyzh) [reply src/docfx/build/toc/TableOfContentsMap.cs]: also assert `_toc.Count == tocs.Count`

Comment (OsmondJiang) [reply src/docfx/build/toc/TableOfContentsMap.cs]: sorry, missing your comment, will change it later

**Pull request merged**